artworks/views.py

```diff
@@ -1,3 +1,64 @@
-from django.shortcuts import render
+from django.shortcuts import render, redirect, reverse
+from .forms import ArtForm
+from django.db import transaction
+from .models import Category, Artwork, ArtLike
+from django.http import HttpResponseRedirect
+from django.contrib.auth import authenticate
+
+
+def upload_view(request):
+    if request.method == "POST":
+        form = ArtForm(request.POST, request.FILES)
+        if form.is_valid():
+            with transaction.atomic():
+                form_save = form.save(commit=False)
+                tags = [tag.lower() for tag in form.cleaned_data['category'].replace(' ', '').split(',')]
+                form_save.uploader = request.user
+                form_save.save()
+                for tag in tags:
+                    category = Category.objects.filter(name=tag)
+                    if category.exists():
+                        form_save.category.add(category[0])
+                    else:
+                        category = Category.objects.create(name=tag)
+                        category.save()
+                        form_save.category.add(category)
```

Comment:
Please overwrite the `save` method of the form and move this business logic there.